### ISSUE TITLE
Remove transient UNHCR registered field instead use UNHCR status

### ISSUE THREAD

**Issue by cazcam34** [talentbeyondboundaries/tbbtalentv2#900]
Hi <@U3TVD430A> and <@U055YR1QT0A>, I'd like to emphasize that the Yes and No buttons in the mini intake for UNHCR registration aren't being directly updated from the mini intake itself. Editing needs to be done by opening the General tab. Thank you in advance! FYI <!subteam^S06MQJ4GA8P|@globalsourcerecruitment>

[Slack Message](https://refugeejobsmarket.slack.com/archives/C42S4AJ73/p1713331327977419?thread_ts=1713331327.977419&cid=C42S4AJ73)

**Comment by cazcam34** [comment]
Discussed with @camerojo our thoughts below:
- Currently as per Ribale's message the field that is being updated (Is candidate registered with unhcr? Yes/No/Unsure) is actually just a transient field that is used to populate the unhcr status field. So this isn't being saved to the DB or able to be reflected elsewhere on the DB.
- The zombie column unhcrRegistered is still on the Candidate table, so we should remove this. We should also remove the transient field and we can just simply update the unhcrStatus table directly.
- Front end we can set Yes/No fields to the relevant enums, without needing the transient field. 
- Problem with saving the same field with slightly different data/question, is the risk of loosing better more detailed data. Eg. A full intake is ocmpleted and entered unhcr status as Mandate Refugee, however someone goes into the mini intake and updates with Yes. 
- Solution around this is that the Yes/No version of the field can only be changed IF there isn't more detailed data. So only if a Yes/No enum value is saved can it be changed. 
- The enums for unhcrStaus are:
    NoResponse("No response"),
    MandateRefugee("Assessed by UNHCR as a mandate refugee (RSD)"),
    RegisteredAsylum("Registered with UNHCR as asylum seeker"),
    RegisteredStateless("Registered with UNHCR as stateless"),
    RegisteredStatusUnknown("Registered with UNHCR but status unknown") **this is the YES option**
    NotRegistered("Not registered") **This is the NO option**
    Unsure("Unsure") **This is the UNSURE option**
    NA("Not applicable");

**Comment by cazcam34** [comment]
The logic that we are after is actually already in the code - only update unhcr status if there is nothing in the status already. So when changing unhcr registered (yes no transient field) we only want to update with the yes no unhcr status if there is no status. If a status already exists, changing yes/no won't do anything. This is currently in the candidate portal, a candidate that tries to change their registration from yes to no won't be allowed to do so ONLY unless they previously didn't have any status (first change) or they put unsure.

**Comment by cazcam34** [comment]
Argh we use UNHCR Registered (simple yes/no/unsure) in SF!!! So can't delete this... yet

**Comment by cazcam34** [comment]
We still use UNHCR Registered for populating in the General tab also... we use the unhcrStatus value and convert that to a Yes/No value. However we can't go from a Yes/No value to a unhcrStatus without the risk of loosing data.

**Comment by cazcam34** [comment]
Updated the component for mini & full to both show the same unhcr status question and no longer having the unhcrRegistered question. Yes users will have more selections in the mini intake (not a simple yes no) but it means we can avoid alot of unneccessary code to have two seperate form controls depending on a full/mini intake which just updates the same thing on the back end anyway. This way it is transparent what the user is updating.
On the Candidate Portal this will need to be different as candidates need a simplified question as they likely won't know the exact status of their registration. In this case I think just having a simple ng-select with customised dropdown <options> or we can just have a different enum which is simplified.

**Comment by cazcam34** [comment]
Code ready - now intakes just display unhcr status and no where on admin portal shows a unhcr registered question.... Candidate portal does use this unhcr registered question as reasons in comment above.
Unhcr registered field is also used and synced to SF.

Database column for unhcr registered has old data saved - around 46k is not null. But over 160k is null (since the we stopped populating the field and made it transient). We ran a flyway to translate this data over to new unhcr status field so it should be safe to delete. just get your thoughts?
update candidate set unhcr_status =
                         case
                             when unhcr_registered = 'No' then 'NotRegistered'
                             when unhcr_registered = 'Yes' then 'RegisteredStatusUnknown'
                             when unhcr_registered = 'NoResponse' then 'NoResponse'
                             when unhcr_registered = 'Unsure' then 'Unsure'
                             end
where unhcr_registered is not null and
    (unhcr_status is null or unhcr_status in ('NoResponse', 'Unsure'));

**Comment by cazcam34** [comment]
Waiting on Ribale response - do deletion of unhcr registered transient field from database another time. Create an issue for it (#906)

**Comment by cazcam34** [comment]
I got two thumbs up from Ribale and Tiffany - take this as a yes?!

**Comment by cazcam34** [comment]
Update the tests?

**Comment by cazcam34** [comment]
Tagged Mariam in thread to get her approval
